manifests/source.pp

```diff
@@ -18,7 +18,7 @@
   $key_content       = undef,
   $key_source        = undef,
   $trusted_source    = undef,
-  $notify_update     = undef,
+  $notify_update     = true,
```

Comment:
in source.pp -> notify_update is passed to `apt::setting` with value `undef`
in setting.pp ->notify_update is `undef`, `validate_bool` there fails because the value of  notify_update is neither true nor false

--> the default for `notify_update` in `source` should not be `undef`, but `true` as the default in `setting`